src/TLNetwork.ts

```diff
@@ -195,17 +182,34 @@ export class TLNetwork {
     this.provider = provider
   }
 
-  public setSigner(signer: TLSigner): void {
-    if (!(signer instanceof Web3Signer || signer instanceof EthersWallet)) {
+  public setSigner(web3Provider, wallet): void {
```

Comment:
please use typings.
also I think it should be `signer` nor `wallet`

Reply:
If I use typing and signer, I will have: `public setSigner(web3Provider, signer: TLSigner)` and will not be able to pass the wallet (that is actually also a signer) to the function
If I use  `public setSigner(web3Provider, signer: TLWallet)`, I cannot do `this.signer = signer` because `this.signer` is of type `TLSigner`.

To solve the problem, since every wallet is a signer I would change the interface `TLWallet` to extend `TLSigner`. This way, we can pass a wallet to a function that expects a signer.

I wanted to do this as part of https://github.com/trustlines-network/clientlib/issues/205 but can do it here if you are find with this approach.

Reply:
if it is a signer, you should be able to pass it.

Reply:
Ah, got it. No you can seperate that.